source/game.ingameinterface.bmx

```diff
@@ -1079,6 +1003,244 @@ Function GetInGameInterface:TInGameInterface()
+								result :+ ["manager"+suffix]
+							Case TVTTargetGroup.Pensioners
+								result :+ ["pensioner"+suffix]
+							Default
+								Throw "unknown audience "+ entry.Key
+						EndSelect
+					EndIf
+				Next
+			EndIf
+		EndIf
+
+rem
+		'add feedback to watching members
+		'leaving original code for later reference
+		Local feedback:TBroadcastFeedback = GetBroadcastManager().GetCurrentBroadcast().GetFeedback(playerID)
+
+		if not feedback or not feedback.AudienceInterest
+			print "Interface.GetWatchingFamily: no AudienceInterest!"
+			debugstop
+		endif
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Children) > 0)
+
+			'maybe sent to bed ? :D
+			'If GetWorldTime().GetDayHour() >= 5 and GetWorldTime().GetDayHour() < 22 then 'manuel: muss im Feedback-Code geprüft werden.
+			result :+ ["girl"]
+		endif
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Housewives) > 0) then result :+ ["mother"]
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Pensioners) > 0) then result :+ ["grandpa"]
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Teenagers) > 0)
+			'in school monday-friday - in school from till 7 to 13 - needs no sleep :D
+			'If GetworldTime().GetWeekday()>6 or (GetWorldTime().GetDayHour() < 7 or GetWorldTime().GetDayHour() >= 13) then result :+ ["teen"] 'manuel: muss im Feedback-Code geprüft werden.
+			result :+ ["teen"]
+		endif
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Unemployed) > 0)
+			result :+ ["unemployed"]
+		else
+			'if there is some audience, show the sleeping unemployed
+			Local audienceResult:TAudienceResult = GetBroadcastManager().GetAudienceResult( GetPlayerBaseCollection().playerID )
+			If audienceResult and audienceResult.GetAudienceQuotePercentage() > 0.05
+				result :+ ["unemployed.bored"]
+			endif
+		endif
+
+		if (feedback.AudienceInterest.GetTotalValue(TVTTargetGroup.Manager) > 0) then result :+ ["manager"]
+endrem
+
+		_assignSpots(result, playerID-1)
+		currentBroadCast[playerID-1] = material
+		currentChannel = playerID
+	End Method
+
+	'assign couch seats
+	Method _assignSpots(newViewers:String[], playerIndex:Int)
+		Local oldMembers:String[] = watchingMembers[playerIndex]
+		Local oldCouchPositions:Int[] = couchPositions[playerIndex]
+		Local finalMembers:String[]
+		Local familyMembersUsed:int = newViewers.length
+		Local newCouchPositions:Int[] = new Int[newViewers.length]
+		'default couch positions
+		If familyMembersUsed >= 3 Then newCouchPositions = [550, 610, 670]
+		If familyMembersUsed = 2 Then newCouchPositions = [580, 640]
+		If familyMembersUsed = 1
+			'prevent alternating of single viewers in center seat
+			newCouchPositions = [610]
+			If oldMembers.length = 1 And oldCouchPositions[0] = 610 Then newCouchPositions = [670]
+		EndIf
+
+		Local unemployedPresent:Int = False
+		Local sameViewerExists:Int = False
+
+		'unemployed always on the left
+		For Local j:int=0 to newViewers.length-1
```

Comment:
Braucht jetzt keine Codeaenderung aber du kannst auch `until` nehmen:
`For Local j:Int = 0 until newViewers.length`.